tools/jdk/BUILD.java_tools

```diff
@@ -15,203 +17,124 @@ SUPRESSED_WARNINGS = select({
-    bootclasspath = ["@bazel_tools//tools/jdk:platformclasspath"],
-    forcibly_disable_header_compilation = 0,
-    genclass = [":GenClass"],
-    header_compiler = [":Turbine"],
-    header_compiler_direct = [":TurbineDirect"],
-    ijar = [":ijar"],
-    jacocorunner = ":jacoco_coverage_runner_filegroup",
-    javabuilder = [":JavaBuilder"],
-    javac = [":javac_jar"],
-    javac_supports_workers = 1,
     jvm_opts = [
         # In JDK9 we have seen a ~30% slow down in JavaBuilder performance when using
         # G1 collector and having compact strings enabled.
         "-XX:+UseParallelOldGC",
         "-XX:-CompactStrings",
-        # Allow JavaBuilder to access internal javac APIs.
-        "--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED",
-        "--add-exports=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED",
-        "--add-exports=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED",
-        "--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED",
-        "--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED",
-        "--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED",
-        "--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED",
-        "--add-opens=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED",
-
-        # override the javac in the JDK.
-        "--patch-module=java.compiler=$(location :java_compiler_jar)",
-        "--patch-module=jdk.compiler=$(location :jdk_compiler_jar)",
-
-        # quiet warnings from com.google.protobuf.UnsafeUtil,
-        # see: https://github.com/google/protobuf/issues/3781
-        # and: https://github.com/bazelbuild/bazel/issues/5599
-        "--add-opens=java.base/java.nio=ALL-UNNAMED",
-        "--add-opens=java.base/java.lang=ALL-UNNAMED",
-    ],
-    misc = [
-        "-XDskipDuplicateBridges=true",
-        "-g",
-        "-parameters",
-    ],
-    singlejar = [":singlejar"],
+    ] + JDK9_JVM_OPTS,
+)
+
+java_toolchain_default(
+    name = "toolchain_hostjdk8",
+    jvm_opts = JDK8_JVM_OPTS,
+    source_version = "8",
+    target_version = "8",
+)
+
+# Default to the Java 8 language level.
+# TODO(cushon): consider if/when we should increment this?
+java_toolchain_default(
+    name = "legacy_toolchain",
+    source_version = "8",
+    target_version = "8",
+)
+
+# Needed for openbsd
+java_toolchain_default(
+    name = "legacy_toolchain_jvm8",
+    jvm_opts = JDK8_JVM_OPTS,
     source_version = "8",
     target_version = "8",
-    tools = [
-        ":java_compiler_jar",
-        ":jdk_compiler_jar",
-    ],
 )
 
+# The 'vanilla' toolchain is an unsupported alternative to the default.
+#
+# It does not provider any of the following features:
+#   * Error Prone
+#   * Strict Java Deps
+#   * Header Compilation
+#   * Reduced Classpath Optimization
+#
+# It uses the version of javac from the `--host_javabase` instead of the
+# embedded javac, which may not be source- or bug-compatible with the embedded
```

Comment:
Do we actually mean "embedded" here, or "local" instead?

Reply:
This is referring to javac which is usually provided with java_tools. I reworded it.